deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -0,0 +1,449 @@
+
+import com.hp.hpl.jena.rdf.model.Bag;
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.NodeIterator;
+import com.hp.hpl.jena.rdf.model.Property;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.rdf.model.Statement;
+import com.hp.hpl.jena.vocabulary.DC;
+import com.hp.hpl.jena.vocabulary.RDF;
+
+import edu.unc.lib.deposit.work.AbstractDepositJob;
+import edu.unc.lib.deposit.work.DepositGraphUtils;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.FolderObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.fedora.FedoraException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.CdrDeposit;
+import edu.unc.lib.dl.reporting.ActivityMetricsClient;
+import edu.unc.lib.dl.util.DepositException;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositField;
+
+/**
+ * Ingests all content objects in the deposit into the Fedora repository.
+ * 
+ * @author bbpennel
+ *
+ */
+public class IngestContentObjectsJob extends AbstractDepositJob {
+	private static final Logger log = LoggerFactory.getLogger(IngestContentObjectsJob.class);
+
+	private boolean resumed;
+
+	// The set of object paths from this deposit which already exist in the repository
+	private Set<String> previouslyIngestedSet;
+
+	@Autowired
+	private ActivityMetricsClient metricsClient;
+
+	public IngestContentObjectsJob() {
+		super();
+	}
+
+	public IngestContentObjectsJob(String uuid, String depositUUID) {
+		super(uuid, depositUUID);
+	}
+
+	/**
+	 * Determines how many operations will be required to complete this deposit.
+	 * 
+	 * @param depositBag
+	 */
+	private void calculateWorkRemaining(Bag depositBag) {
+		List<String> contentPaths = new ArrayList<>();
+		DepositGraphUtils.walkChildrenDepthFirst(depositBag, contentPaths, true);
+		// Number of actions is the number of ingest objects plus deposit record
+		setTotalClicks(contentPaths.size());
+
+		// Deposit is restarting from part way through
+		resumed = getDepositStatusFactory().isResumedDeposit(getDepositUUID());
+		if (resumed) {
+			populatePreviouslyIngested(contentPaths);
+		}
+	}
+
+	/**
+	 * Generates a list of objects in this deposit which already exist in the
+	 * repository, updating the status of the job to indicate how much work is
+	 * left
+	 * 
+	 * @param contentPaths
+	 */
+	private void populatePreviouslyIngested(List<String> contentPaths) {
+		// Generate a list of object paths from this deposit already in fedora
+		previouslyIngestedSet = new HashSet<>();
+		for (String path : contentPaths) {
+			if (repository.objectExists(PIDs.get(path))) {
+				previouslyIngestedSet.add(path);
+			}
+		}
+		// Update the ingested object count to reflect the number in the repo
+		getJobStatusFactory().setCompletion(getJobUUID(), previouslyIngestedSet.size());
```

Comment:
No, it is instead starting from the number of already ingested items and keeping the total number of objects the same.  My hope is that this will result in a more consistent count in the status monitor, although might need to see it in action to be sure